apps/service_providers/llm_service/history_managers.py

```diff
@@ -64,12 +65,12 @@ def for_llm_chat(
             experiment=experiment,
             max_token_limit=experiment.max_token_limit,
             chat_model=experiment.get_chat_model(),
-            trace_service=trace_service or experiment.trace_service,
+            trace_service=trace_service,
```

Comment:
The param list indicates that `trace_service` might be `None`, but this is not the case anymore, since it will only be a passthrough when the underlying trace service is `None`?

Reply:
It can be None in tests though we could change that.